src/service/navigation.js

```diff
@@ -566,6 +567,10 @@ export class Navigation {
       ) {
         this.removeViewerQueryBeforeNavigation_(win, fromLocation, target);
       }
+
+      if (this.navigateToUrlWithViewer(to, 'external nav')) {
```

Comment:
This is now returning a promise instance (which is truthy), which means we'll always call `preventDefault`.

Reply:
Updated - we're handling the async calls in the constructor so that the calls are (most likely) finished by the time this function is called, and it can once again return a boolean.